plugins/modules/apt_sources_facts.py

```diff
@@ -0,0 +1,160 @@
+description:
+  - Set configured apt sources as facts.
+  - Requires the C(python-apt) or C(python3-apt) package to be installed.
+version_added: "1.0.2"
+author: Adfinis AG (@adfinis)
+options: {}
+'''
+
+
+EXAMPLES = r'''
+- name: Populate apt sources facts
+  adfinis.facts.apt_sources_facts:
+
+- name: Print apt sources facts
+  ansible.builtin.debug:
+    var: ansible_facts.apt_sources
+'''
+
+
+RETURN = r'''
+ansible_facts:
+  description: Facts to add to ansible_facts about the configured apt sources.
+  returned: always
+  type: dict
+  contains:
+    apt_sources:
+      description: List of all active apt sources, both from deb822 and legacy format files.
+      returned: always
+      type: list
+      elements: dict
+      sample:
+        - file: /etc/apt/sources.list.d/debian.sources
+          types: [deb, deb-src]
+          uri: http://deb.debian.org/debian
+          suites: [boomworm, bookworm_updates]
+          components: [main, non-free-firmware]
+          architectures: []
+        - file: /etc/apt/sources.list.d/debian.sources
+          types: [deb, deb-src]
+          uri: http://security.debian.org/debian-security
+          suites: [bookworm-security]
+          components: [main, non-free-firmware]
+          architectures: []
+      contains:
+        file:
+          description: Name of the file this entry was found in.
+          returned: always
+          type: str
+          sample: /etc/apt/sources.list
+        types:
+          description: Types of the entry, contains "deb", "deb-src" or both.
+          returned: always
+          type: list
+          elements: str
+          sample:
+            - deb
+            - deb-src
+        uri:
+          description: URI for the repository.
+          returned: always
+          type: str
+          sample: http://deb.debian.org/debian
+        suites:
+          description: Name of the suites of the repository.
+          returned: always
+          type: list
+          elements: str
+          sample:
+            - bookworm
+            - bookworm-updates
+        components:
+          description: Enabled components of the repository.
+          returned: always
+          type: list
+          elements: str
+          sample:
+            - main
+            - non-free-firmware
+        architectures:
+          description: The entries of the "Architectures" option, if any.
+          returned: always
+          type: list
+          elements: str
+          sample:
+            - amd64
+'''
+
+
+def main():
+    module = AnsibleModule(argument_spec=dict(), supports_check_mode=True)
+
+    # Import the sourceslist module from python-apt; emit a warning and exit if not possible.
+    try:
+        from aptsources.sourceslist import SourcesList, Deb822SourceEntry, SourceEntry
+    except ImportError:
+        warnings = [
+            'adfinis.facts.apt_sources_facts was invoked on a system that is missing the python3-apt or python-apt' +
```

Comment:
I think that as it is, this will result in a missing space:
"missing the python3-apt or python-aptpackage.  ansible_facts.apt_sources will be empty."